idunn/utils/prometheus.py

```diff
@@ -0,0 +1,69 @@
+import time
+
+from apistar import Component
+from prometheus_client import Counter, Gauge, Histogram
+
+
+WIKI_REQUEST_DURATION = Histogram(
+    "wiki_request_duration_seconds",
+    "Time spent processing a Wiki request.",
+    ["handler"],
+)
+
+WIKI_BREAKER_STATUS = Gauge(
+    "breaker_status",
+    "Breaker status, open (=2) or half-open (=1) or close (=0).",
+    []
+)
+
+WIKI_BREAKER_EXCEPTIONS_COUNT = Counter(
+    "breaker_exceptions_count",
+    "Number of exceptions caught by the breaker.",
+    []
+)
+
+WIKI_RATE_LIMITER_MAX_COUNT = Counter(
+    "limiter_too_many_requests_count",
+    "Number of times TooManyRequests exception has been raised.",
+    []
+)
+
+BREAKER_STATUS = {'closed': 0, 'half-open':1, 'open': 2}
+
+class PrometheusTracker:
+    """ The logic of the Prometheus metrics is defined in this component """
+
+    _tracker = None
+
+    class _Prometheus:
+        """ Singleton that catch some primitives on the Prometheus metrics """
+
+        def request_start(self, handler):
```

Comment:
I think the split `request_start`/`request_end` method can lead to many errors, a context manager would be better

```python

    @contextlib.contextmanager
    def log_request_duration(self, handler):
        with WIKI_REQUEST_DURATION.labels(handler).time():
            yield
```

Reply:
indeed :+1: